appserver/transaction/jta/src/main/java/com/sun/enterprise/transaction/JavaEETransactionManagerSimplified.java

```diff
@@ -109,11 +112,13 @@
     @Inject protected InvocationManager invMgr;
 
     @Inject private Provider<RequestTracingService> requestTracing;
-    
+
+    @Inject private PayaraExecutorService payaraExecutorService;
```

Comment:
I don't think Payara Executor Service should be used for this. This is core application processing logic and I feel this could overload the Payara Executor Service which was planned for more internal tasks. For example if you have a web application performing 100s of requests per second with timeouts enabled we could be adding and removing 100s of timeout tasks per second on this Executor Pool. I approve of replacing timer with the more modern executor service though.

Reply:
Replaced PayaraExecutorService with newly created ScheduledThreadPoolExecuter